content/blog/getting-started-with-volume-snapshots-on-a-kubernetes-cluster-in-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,840 @@
+```
+ 
+Now that you understand the basics, in﻿ the following sections, I will describe how to create volume snapshots of persistent volumes in K8s using the HPE CSI driver for K8s.  
+
+### Prerequisites
+
+
+
+Before starting, make sure you have the following:
+
+
+
+
+
+* A K8s cluster, being provisioned in HPE GreenLake for Private Cloud Enterprise
+* The kubectl CLI tool, together with the kubeconfig file for accessing the K8s cluster
+* The o﻿ptional mysql CLI tool, for accessing the deployed sample MySQL database service
+
+### Deploy MySQL database
+
+
+
+B﻿efore showing the volume snapshots, a MySQL database instance from [my GitHub repo](https://github.com/GuopingJia/mysql-app) will be deployed as a sample stateful application to the cluster. 
+
+
+**1﻿. Install MySQL database**
+
+
+
+MySQL database requires a persistent volume to store data. Here is the PVC YAML manifest file *mysql-pvc.yaml* in the repo's *base* folder:
+
+
+
+```markdown
+
+
+$ cat mysql-app/base/mysql-pvc.yaml 
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: mysql-pvc
+  namespace: mysql
+  labels:
+    app: mysql
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+
+```
+
+
+
+This PVC file, together with other YAML manifest files in the folder *base*, will be used to install a MySQL database instance using [Kustomize](https://kustomize.io/). 
+
+
+
+```markdown
+ $ tree mysql-app/base
+mysql-app/base
+├── kustomization.yaml
+├── mysql-deployment.yaml
+└── mysql-pvc.yaml
+```
+
+
+
+The file *kustomization.yaml* lists all YAML files in its resources section, together with the secret generator for MySQL password:
+
+
+
+```markdown
+$ cat mysql-app/base/kustomization.yaml
+secretGenerator:
+- name: mysql-pass
+  namespace: mysql
+  literals:
+  - password=CfeDemo@123
+resources:
+  - mysql-deployment.yaml
+  - mysql-pvc.yaml
+
+```
+
+
+
+T﻿ype below command to install the MySQL database to the namespace *mysql*:
+
+
+
+```shell
+$ kubectl apply -k mysql-app/base
+namespace/mysql created
+secret/mysql-pass-m62cbhd9kf created
+service/mysql created
+persistentvolumeclaim/mysql-pvc created
+deployment.apps/mysql created
+
+
+```
+
+T﻿ype the following command to check the MySQL database deployment state. The MySQL Pod should be in *Running* status:
+
+```shell
+$ kubectl get all -n mysql
+NAME                         READY   STATUS    RESTARTS   AGE
+pod/mysql-6974b58d48-wb8g5   1/1     Running   0          14s
+
+NAME            TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)    AGE
+service/mysql   ClusterIP   None         <none>        3306/TCP   24s
+
+NAME                    READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/mysql   1/1     1            1           23s
+
+NAME                               DESIRED   CURRENT   READY   AGE
+replicaset.apps/mysql-6974b58d48   1         1         1       24s
+```
+
+
+
+Y﻿ou can check that the PVC and the PV created as part of MySQL database deployment:
```

Comment:
Hmmm. A bit confusing. Do you mean this?
"You can check that the PVC and the PV **were** created as part of **the** MySQL database deployment:"